src/Compiler/Driver/GraphChecking/Graph.fsi

```diff
@@ -0,0 +1,21 @@
+namespace FSharp.Compiler.GraphChecking
+
+open System.Collections.Generic
+
+/// A Directed Acyclic Graph (DAG) of arbitrary nodes.
+type internal Graph<'Node> = IReadOnlyDictionary<'Node, 'Node[]>
+
+/// Functions for operating on the Graph type.
+module internal Graph =
+    /// Build the graph.
+    val make: nodeDeps: seq<'Node * 'Node array> -> IReadOnlyDictionary<'Node, 'Node array> when 'Node: equality
```

Comment:
Should the return type be `Graph` here? It would look better in type hints at the call site.
```suggestion
    val make: nodeDeps: seq<'Node * 'Node array> -> Graph<'Node> when 'Node: equality
```